modules/querier/http.go

```diff
@@ -36,10 +46,24 @@ func (q *Querier) TraceByIDHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// validate request
+	valid, blockStart, blockEnd, queryIngesters := validateAndSanitizeRequest(r)
+	if !valid {
+		http.Error(w, "invalid parameters", http.StatusBadRequest)
```

Comment:
it would be nice if `validateAndSanitizeRequest` returned an `error` with the reason for the failed validation which we then returned in the body.